src/ui/public/agg_types/param_types/base.js

```diff
@@ -6,5 +6,15 @@ export function BaseParamTypeProvider() {
     _.assign(this, config);
   }
 
+  /**
+   *  A function that will be called before an aggConfig is serialized and sent to ES.
+   *  Allows aggConfig to retrieve values needed for serialization by creating a {SearchRequest}
+   *  Example usage: an aggregation needs to know the min/max of a field to determine an appropriate interval
+   *
+   *  @returns {Promise<undefined>|undefined}
+   */
+  BaseParamType.prototype.onSearchRequestStart = function () {
```

Comment:
I updated the method name and added `@param` comments about the arguments but I could not put the arguments in the method definition since BaseParamType#modifyAggConfigOnSearchRequestStart does not use the arguments and eslint is not happy about unused arguments